File: 13 opencv（四）/c++/hist_eq.cpp
#include<opencv2/opencv.hpp>

using namespace std;

int main() {
    cv::Mat img = cv::imread("7.jpg", cv::IMREAD_GRAYSCALE);

    cv::Mat dst;
    cv::equalizeHist(img, dst);
    cv::imshow("dst", dst);
    cv::waitKey(0);
}

File: 13 opencv（四）/hist_eq.py
import cv2
import matplotlib.pyplot as plt

img = cv2.imread('7.jpg', 0)
cv2.imshow("src", img)

his = cv2.calcHist([img], [0], None, [256], [0, 255])
plt.plot(his, label='his', color='r')

dst = cv2.equalizeHist(img)
cv2.imshow("dst", dst)

his = cv2.calcHist([dst], [0], None, [256], [0, 255])
plt.plot(his, label='his', color='b')
plt.show()

cv2.waitKey()